tests/system/src/test/java/test/javafx/scene/text/ArabicWrappingTest.java

```diff
@@ -0,0 +1,214 @@
+
+    static class SystemErrFilter extends PrintStream {
+        private boolean foundException = false;
+        private String exceptionMsg;
+
+        public SystemErrFilter(OutputStream out) {
+            super(out);
+        }
+
+        @Override
+        public synchronized void print(String s) {
+            System.out.flush();
+            if (s.indexOf("Exception") >= 0) {
+                foundException = true;
+                exceptionMsg = s;
+            }
+            super.print(s);
+        }
+
+        boolean checkException() {
+            return foundException;
+        }
+
+        String getExceptionString() {
+            return exceptionMsg;
+        }
+    }
+
+    @BeforeClass
+    public static void initFX() {
+        systemErrFilter = new SystemErrFilter(System.err);
+        System.setErr(systemErrFilter);
+        new Thread(() -> {
+            Application.launch(ArabicWrappingTest.class);
+        }).start();
+    }
+
+    @AfterClass
+    public static void exitTest() {
+        Platform.exit();
+    }
+
+    static volatile boolean testDone = false;
+    static volatile boolean testPassed;
+
+    @Test
+    public void testWrapping() {
+
+       while (!ArabicWrappingTest.testDone) {
+           try {
+               Thread.sleep(2000);
+           } catch (Exception e) {
+           }
+       }
+
+       assertTrue(ArabicWrappingTest.testPassed);
+    }
+
+    public static void main(String[] args) {
+        initFX();
+        try {
+            wrappingTest.testWrapping();
+        } catch (Throwable e) {
+            e.printStackTrace();
+        } finally {
+            exitTest();
+        }
+    }
+
+    static final int MAX_WW = 200;
+    static final int MIN_WW =   5;
+
+    static String text = "Arabic \u0643\u064e\u062a\u064e\u0628\u064e\u200e kataba.";
+    static ArabicWrappingTest wrappingTest;
+
+    @Override
+    public void start(Stage stage) throws Exception {
+        wrappingTest = this;
+        fontNames = Font.getFontNames().toArray(new String[0]);
+        maxFonts = fontNames.length > MAXFONTS ? MAXFONTS : fontNames.length;
+        String[] tmpFonts = new String[maxFonts+4];
+        tmpFonts[0] = "System Regular";
+        tmpFonts[1] = "SansSerif Regular";
+        tmpFonts[2] = "Serif Regular";
+        tmpFonts[3] = "Monospaced Regular";
+        System.arraycopy(fontNames, 0, tmpFonts, 4, maxFonts);
+        fontNames = tmpFonts;
+        maxFonts = fontNames.length;
+        textNode = new Text(text);
+        font = new Font(fontNames[fontIndex++], 12);
+        System.out.println(font); System.out.flush();
+        textNode.setFont(font);
+        textNode.setWrappingWidth(MAX_WW);
+
+        HBox hbox = new HBox();
+        hbox.getChildren().addAll(textNode);
+        Scene scene = new Scene(hbox);
+        stage.setScene(scene);
+        stage.setTitle("Test bidi text wrapping");
+        stage.setWidth(MAX_WW+50);
+        stage.setHeight(600);
```

Comment:
In connection with `Util.launch(startupLatch, ...)`, here is where you can add:

```
    stage.setOnShown(e -> Platform.runLater(startupLatch::countDown));
```

Reply:
Because of the sleep loop in my test I don't need this.